src/System.Globalization/tests/PlatformDetection.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Diagnostics;
+using System.IO;
+
+public static class PlatformDetection
+{
+    private static bool? s_isUbuntu1510;
+
+    public static bool IsUbuntu1510
+    {
+        get
+        {
+            if (!s_isUbuntu1510.HasValue)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                {
+                    IdVersionPair v = ParseOsReleaseFile();
+
+                    if (v.Id == "ubuntu" && v.VersionId == "15.10")
+                    {
+                        s_isUbuntu1510 = true;
+                        return true;
+                    }
+                }
+
+                s_isUbuntu1510 = false;
+            }
+
+            return s_isUbuntu1510.Value;
+        }
+    }
+
+    private static IdVersionPair ParseOsReleaseFile()
+    {
+        Debug.Assert(RuntimeInformation.IsOSPlatform(OSPlatform.Linux));
+
+        IdVersionPair ret = new IdVersionPair();
+
+        foreach (string line in File.ReadLines("/etc/os-release"))
+        {
+            if (line.StartsWith("ID=", System.StringComparison.Ordinal))
+            {
+                ret.Id = line.Substring("ID=".Length);
+            }
+            else if (line.StartsWith("VERSION_ID=", System.StringComparison.Ordinal))
+            {
+                ret.VersionId = line.Substring("VERSION_ID=".Length);
+            }
+        }
+
+        string versionId = ret.VersionId;
+
+        if (versionId != null && versionId.Length >= 2 && versionId[0] == '"' && versionId[versionId.Length - 1] =='"')
+        {
+            // Remove Quotes.
+            ret.VersionId = versionId.Substring(1, versionId.Length - 2);
+        }
+
+        return ret;
```

Comment:
On my Ubuntu 14.04 machine, this program:

```
using System;
using System.Runtime.InteropServices;

namespace ConsoleApplication
{
    public class Program
    {
        public static void Main(string[] args)
        {
            Console.WriteLine(RuntimeInformation.OSDescription);
        }
    }
}
```

Writes just:

`Linux 3.13.0-53-generic #89-Ubuntu SMP Wed May 20 10:34:39 UTC 2015`